lib/basedriver/helpers.js

```diff
@@ -34,32 +35,25 @@ const APPLICATIONS_CACHE_GUARD = new AsyncLock();
 const SANITIZE_REPLACEMENT = '-';
 const DEFAULT_BASENAME = 'appium-app';
 
-
-async function cleanupCachedApps () {
+process.on('exit', () => {
   if (!APPLICATIONS_CACHE.length) {
     return;
   }
+  const appPaths = APPLICATIONS_CACHE.values()
+    .map(({fullPath}) => fullPath);
+  APPLICATIONS_CACHE.reset();
 
-  logger.debug('Performing cleanup of the applications cache');
-  const cleanupPromises = [];
-  for (const {fullPath} of APPLICATIONS_CACHE.values()) {
-    cleanupPromises.push((async () => {
-      if (await fs.exists(fullPath)) {
-        await fs.rimraf(fullPath);
-      }
-    })());
-  }
-  try {
-    await B.all(cleanupPromises);
-  } catch (e) {
-    logger.error(e);
-  } finally {
-    APPLICATIONS_CACHE.reset();
+  logger.debug(`Performing cleanup of ${appPaths.length} cached ` +
+    `application${appPaths.length > 1 ? 's' : ''}`);
```

Comment:
This will give the wrong grammar if there are `0` applications.

Reply:
%), makes sense

Reply:
however in this particular case `appPaths.length` will always be greater than zero because of the preceding guard condition